### PR TITLE
Test setting rpath for release build

### PR DESCRIPTION
**Pull request by fasmat**
- Add workflow_dispatch for release workflow
- Adds settings to `Cargo.toml` to try and fix `rpath` issues
- Adds [`strip`](https://doc.rust-lang.org/rustc/codegen-options/index.html#strip) and [`lto`](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) to cargo profile as well

### CODE REVIEW

**Comment by fasmat** [comment]
Closed, as this change should be included in another PR. See https://github.com/spacemeshos/spacemesh-sdk/pull/4#discussion_r1205511318

**Comment by lrettig** [comment]
I am actually running into https://github.com/spacemeshos/smkeys/issues/6 as well. Unfortunately the fixes in this branch don't fix the issue for me.

**Comment by lrettig** [comment]
Fixed downstream in https://github.com/spacemeshos/smkeys/pull/8; assuming that works/is merged, this can be closed

**Comment by fasmat** [comment]
As mentioned in https://github.com/spacemeshos/smkeys/pull/8 I would prefer fixing it upstream to avoid having to set `DYLD_LIBRARY_PATH` explicitly for every user of spacemesh-sdk. What do you think?

**Comment by lrettig** [comment]
See above comment: https://github.com/spacemeshos/spacemesh-sdk/pull/5/files#r1210577136

Based on my limited understanding, I don't see any way to fix the issue on this end using rpath. It seems it needs to be sorted out downstream since only the calling library knows the location of the dylib relative to the executable in question.

@fasmat if you or anyone else can come up with a working solution then I'm all ears. But for now I think we need to go with something like https://github.com/spacemeshos/smkeys/pull/8.

**Comment by fasmat** [comment]
I added a comment with a solution that seems to work: https://github.com/spacemeshos/spacemesh-sdk/pull/5#discussion_r1210601572

File: Cargo.toml
Comment (lrettig) [reply Cargo.toml:30]: How is this better than setting `DYLD_LIBRARY_PATH`? (Not a rhetorical question.)

Comment (fasmat) [reply Cargo.toml:30]: From my (limited) understanding `DYLD_LIBRARY_PATH` should not be set explicitly and seems to not be forwarded to child processes for security reasons. (Sources: https://briandfoy.github.io/macos-s-system-integrity-protection-sanitizes-your-environment/ and https://blog.krzyzanowskim.com/2018/12/05/rpath-what/).

Afaik `DYLD_LIBRARY_PATH` also changes the lookup paths of where `go` looks for its dependencies, possibly altering the behaviour of the compiler.

Instead it seems to be recommended to use `LD_FLAGS` / `CGO_LDFLAGS` instead. As additional side effect the libraries built from `spacemesh-sdk` now don't set the path they were built in any more as lookup path (so we are not leaking info via our libraries any more).

File: .github/workflows/release.yml
Comment (fasmat) [inline .github/workflows/release.yml:69]: Please remove before merge 🙂

**Review by fasmat — COMMENTED** [review]
Thanks for the effort @lrettig! I would approve the PR now, but it also contains commits from me so maybe @poszu can approve if there aren't any objections?

**Review by lrettig — APPROVED** [review]

**Pull request merged**
